src/passes/externImporter.ts

```diff
@@ -60,7 +52,10 @@ export class ExternImporter extends ASTMapper {
     assert(sourceUnit !== undefined, 'Trying to import a definition into an unknown source unit');
     if (declarationSourceUnit === undefined || sourceUnit === declarationSourceUnit) return;
 
-    if (declaration instanceof FunctionDefinition || declaration instanceof StructDefinition) {
+    if (
+      declaration instanceof FunctionDefinition ||
+      (declaration instanceof StructDefinition && isFree(declaration))
```

Comment:
There is previoush check in this function that checks for non-free declarations making `isFree(declaration)` always evaluate to `true`. If it is intended to only apply this condition to structs then that previous check should be removed (line 47)